src/test/java/edu/cornell/kfs/concur/batch/service/impl/MockConcurRequestV4Server.java

```diff
@@ -0,0 +1,221 @@
+package edu.cornell.kfs.concur.batch.service.impl;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.StringUtils;
+import org.kuali.kfs.krad.util.UrlFactory;
+
+import edu.cornell.kfs.concur.ConcurConstants.ConcurApiOperations;
+import edu.cornell.kfs.concur.ConcurConstants.ConcurApiParameters;
+import edu.cornell.kfs.concur.ConcurConstants.RequestV4Status;
+import edu.cornell.kfs.concur.ConcurConstants.RequestV4StatusCodes;
+import edu.cornell.kfs.concur.ConcurConstants.RequestV4Views;
+import edu.cornell.kfs.concur.ConcurUtils;
+import edu.cornell.kfs.concur.batch.fixture.ConcurFixtureUtils;
+import edu.cornell.kfs.concur.batch.service.impl.fixture.RequestV4DetailFixture;
+import edu.cornell.kfs.concur.batch.service.impl.fixture.RequestV4PersonFixture;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4ListItemDTO;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4ListingDTO;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4OperationDTO;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4ReportDTO;
+import edu.cornell.kfs.concur.rest.jsonObjects.ConcurRequestV4StatusDTO;
+
+public class MockConcurRequestV4Server implements Closeable {
+
+    private static final int MAX_RESULT_LIMIT = 100;
+
+    private static final Set<String> ALLOWED_REQUEST_LIST_QUERY_PARAMETERS = Set.of(
+            ConcurApiParameters.VIEW, ConcurApiParameters.START, ConcurApiParameters.LIMIT,
+            ConcurApiParameters.MODIFIED_AFTER, ConcurApiParameters.MODIFIED_BEFORE, ConcurApiParameters.USER_ID);
+
+    private ConcurrentMap<String, RequestEntry> travelRequests;
+    private String baseRequestV4Url;
+
+    public MockConcurRequestV4Server(String baseRequestV4Url, RequestV4DetailFixture... requestsToAdd) {
+        this.travelRequests = new ConcurrentHashMap<>();
+        this.baseRequestV4Url = baseRequestV4Url;
+        addTravelRequests(requestsToAdd);
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (travelRequests != null) {
+            travelRequests.clear();
```

Comment:
is this needed?  Shouldn't setting travelRequests to null clear everything up?

Reply:
I agree that this clear() operation is not needed for the unit testing. I'll remove it.